hta/analyzers/critical_path_analysis.py

```diff
@@ -392,9 +367,19 @@ def critical_path_analysis_for_rank(
         # Consider all events that start within the annotatated window.
         # and also fiter out 0 duration events as they mess up the
         # event stack generation
-        clipped_df = trace_df.query(
-            f"(ts >= {start_ts} and ts <= {end_ts}) and " " dur > 0"
-        ).copy()
+        a = trace_df.query(f"(ts >= {start_ts} and ts <= {end_ts}) and dur > 0")
+
+        # Also consider GPU kernels whose runtime events are in the correct
+        # time window
+        gpu_kernels = trace_df[trace_df["stream"].ne(-1)]
```

Comment:
@anupambhatnagar  this implements the time windowing such that related GPU kernels can still get through if their CPU launch happened in the time window